model/riscv_fetch_rvfi.sail

```diff
@@ -7,10 +7,10 @@
 /*=======================================================================================*/
 
 function rvfi_fetch() -> FetchResult = {
-  rvfi_inst_data[rvfi_order] = minstret;
-  rvfi_pc_data[rvfi_pc_rdata] = zero_extend(get_arch_pc());
-  rvfi_inst_data[rvfi_mode] = zero_extend(privLevel_to_bits(cur_privilege));
-  rvfi_inst_data[rvfi_ixl] = zero_extend(misa[MXL]);
+  rvfi_set_pc_data_rdata(zero_extend(get_arch_pc()));
+  rvfi_set_inst_data_order(minstret);
+  rvfi_set_inst_data_mode(zero_extend(privLevel_to_bits(cur_privilege)));
+  rvfi_set_inst_data_ixl(zero_extend(misa[MXL]));
```

Comment:
I was thinking that instead we could add handy accessors to the Sail model and preserve them, then call them from the RVFI code.

Like

```
function get_ixl() -> bits(2) = misa[MXL]
```

```
--c-preserve get_ixl
```

```
rvfi_inst_data.rvfi_ixl = zget_ixl();
```

Apart from that, and Alexander's comments this looks like almost exactly what I was thinking - nice one!

Eventually I think we could replace `rvfi_get_insn()` with a generic instruction forcing method but that can be in a future PR.